crates/cranelift/src/gc/enabled.rs

```diff
@@ -680,9 +680,12 @@ pub fn translate_array_len(
         // don't know its length yet. Chicken and egg problem.
         BoundsCheck::Access(ir::types::I32.bytes()),
     );
-    let result = builder
-        .ins()
-        .load(ir::types::I32, ir::MemFlags::trusted(), len_field, 0);
+    let result = builder.ins().load(
+        ir::types::I32,
+        ir::MemFlags::trusted().with_readonly(),
```

Comment:
How come this is readonly? Can't the length change over time?

Reply:
Nope, GC arrays have fixed length.